test-suite/Scripting/DuktapeSpec.hs

```diff
@@ -95,3 +98,19 @@ function objTest (obj) { return obj.name + obj.stuff.filter(function (x) { retur
       rE `shouldBe` (Left "TypeError: error (rc -6)")
       rF ← evalDuktape (fromJust ctx) "try { X.add(undefined, \"wtf\") } catch (e) { 0; }"
       rF `shouldBe` (Right $ Just $ Number 0)
+
+  describe "createGovernedHeap" $ do
+    it "applies CheckTimeout to signal the runtime to terminate with RangeError" $ do
+      guvnor ← allowQuarterSecond
+      ctx ← createGovernedHeap nullFunPtr nullFunPtr nullFunPtr guvnor nullFunPtr
+      rE ← evalDuktape (fromJust ctx) "while(true) {};"
+      rE `shouldBe` Left "RangeError: execution timeout"
+
+
+allowQuarterSecond ∷ IO (IO Bool)
```

Comment:
This will add a pretty hard performance penalty over having a separate thread just write a bool and this callback just reading an IORef atomically.

Reply:
That would be a better example to provide to people. Duktape doesn't invoke this callback very frequently so actual cost is maybe not as bad as you're thinking; I did some benchmarking but don't have the data handy now unfortunately.